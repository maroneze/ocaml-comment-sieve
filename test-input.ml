# 1 "test-input.ml"

(*** three asterisks *)
let f () =
  Format.printf "inside f@."

let scan_escape char =
  match char with
  | 'e' | 'E' -> '\027'  (* ASCII code 27. This is a GCC extension *)
  | '\'' -> '\''  (* just a single quote *)
  | '"' -> '"'    (* '"' this causes mayhem on the following lines *)
  | '\027' -> '\027' (* "\027" '\027' *)
  | '\\' -> '\\' (* \\'\\ *)
  | '\x47' -> '\047' (* \\x47 '\\x47' *)
  | other -> (* "\x0f" '\129' *)
    Printf.printf "Unrecognized escape sequence: \\%c" other; (* \\%c *)
    invalid_arg "bla"

let () =
  (* one asterisk (* nested comment *)
                  (** nested comment with two closing asterisks, multilines and "quotes" **) *)
  Format.printf "hello world@.";
  (** two asterisks *)
  f(); (** f is to the left *)
  Format.printf "string with unbalanced (* and escaped double quote (\")@."; (* comment with unbalanced "" inside string and double quotes *)
  Format.printf "escaped backslash inside string: \\@."; (* escaped backslash inside string: "\\" *)
  Format.printf "escaped quote inside string: \"@."; (* escaped quote inside string: "\"" *)
  (* quoted backslash '\\' *)
  (* quoted single quote '\'' *)
  Format.printf "quote chars: %c %c %c@." '"' '\\' '\''; (* several quoted chars *)
  (* lone backslash: \\ *)
  (* lone escaped character: \n *)
  ()
