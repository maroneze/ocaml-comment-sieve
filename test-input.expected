                   

(*** three asterisks *)
          
                            

                      
                 
                         (* ASCII code 27. This is a GCC extension *)
                  (* just a single quote *)
                  (* '"' this causes mayhem on the following lines *)
                     (* "\027" '\027' *)
                 (* \\'\\ *)
                     (* \\x47 '\\x47' *)
             (* "\x0f" '\129' *)
                                                              (* \\%c *)
                     

        
  (* one asterisk (* nested comment *)
                  (** nested comment with two closing asterisks, multilines and "quotes" **) *)
                                
  (** two asterisks *)
       (** f is to the left *)
                                                                             (* comment with unbalanced "" inside string and double quotes *)
                                                         (* escaped backslash inside string: "\\" *)
                                                     (* escaped quote inside string: "\"" *)
  (* quoted backslash '\\' *)
  (* quoted single quote '\'' *)
                                                         (* several quoted chars *)
  (* lone backslash: \\ *)
  (* lone escaped character: \n *)
    
